sparse/mlir_backend/tests/test_simple.py

```diff
@@ -297,3 +299,97 @@ def test_copy():
+        pytest.param("csc", marks=pytest.mark.xfail(reason="https://github.com/llvm/llvm-project/pull/109641")),
+        pytest.param("coo", marks=pytest.mark.xfail(reason="https://github.com/llvm/llvm-project/pull/109135")),
+    ],
+)
+@pytest.mark.parametrize(
+    ("shape", "new_shape"),
+    [
+        ((100, 50), (25, 200)),
+        ((100, 50), (10, 500, 1)),
+        ((80, 1), (8, 10)),
+        ((80, 1), (80,)),
+    ],
+)
+def test_reshape(rng, dtype, format, shape, new_shape):
+    DENSITY = 0.5
+    sampler = generate_sampler(dtype, rng)
+
+    arr_sps = sps.random_array(
+        shape, density=DENSITY, format=format, dtype=dtype, random_state=rng, data_sampler=sampler
+    )
+    arr_sps.eliminate_zeros()
+    arr_sps.sum_duplicates()
+    arr = sparse.asarray(arr_sps)
+
+    actual = sparse.reshape(arr, shape=new_shape)
+    assert actual.shape == new_shape
+
+    try:
+        scipy_format = sparse.to_scipy(actual).format
+    except RuntimeError:
+        pytest.xfail("No library to compare to.")
+
+    expected = sparse.asarray(arr_sps.reshape(new_shape).asformat(scipy_format))
+
+    for x, y in zip(expected.get_constituent_arrays(), actual.get_constituent_arrays(), strict=True):
+        np.testing.assert_array_equal(x, y)
+
+
+@parametrize_dtypes
+def test_reshape_csf(dtype):
+    # CSF
+    csf_shape = (2, 2, 4)
+    csf_format = sparse.levels.get_storage_format(
+        levels=(
+            sparse.levels.Level(sparse.levels.LevelFormat.Dense),
+            sparse.levels.Level(sparse.levels.LevelFormat.Compressed),
+            sparse.levels.Level(sparse.levels.LevelFormat.Compressed),
+        ),
+        order="C",
+        pos_width=64,
+        crd_width=64,
+        dtype=sparse.asdtype(dtype),
+    )
+    for shape, new_shape, expected_arrs in [
+        (
+            csf_shape,
+            (4, 4, 1),
+            [
+                np.array([0, 0, 3, 5, 7]),
+                np.array([0, 1, 3, 0, 3, 0, 1]),
+                np.array([0, 1, 2, 3, 4, 5, 6, 7]),
+                np.array([0, 0, 0, 0, 0, 0, 0]),
+                np.array([1, 2, 3, 4, 5, 6, 7]),
+            ],
+        ),
+        (
+            csf_shape,
+            (2, 1, 8),
+            [
+                np.array([0, 1, 2]),
+                np.array([0, 0]),
+                np.array([0, 3, 7]),
+                np.array([4, 5, 7, 0, 3, 4, 5]),
+                np.array([1, 2, 3, 4, 5, 6, 7]),
+            ],
+        ),
+    ]:
+        arrs = get_example_csf_arrays(dtype)
+        csf_tensor = sparse.from_constituent_arrays(format=csf_format, arrays=arrs, shape=shape)
+
+        result = sparse.reshape(csf_tensor, shape=new_shape)
+        for actual, expected in zip(result.get_constituent_arrays(), expected_arrs, strict=True):
+            np.testing.assert_array_equal(actual, expected)
+
+    # DENSE
+    # NOTE: dense reshape is probably broken in MLIR in 19.x branch
+    # dense = np.arange(math.prod(SHAPE), dtype=dtype).reshape(SHAPE)
```

Comment:
As were are using our wheels (and .20dev LLVM version) we should uncomment and test this case.